src/workerd/api/streams/writable.h

```diff
@@ -152,9 +152,17 @@ class WritableStream: public jsg::Object {
     });
   }
 
+  bool getPipeToCalled() { return pipeToCalled; };
+  // Determines whether this WritableStream had been the target of a pipeTo().
+
+  void setPipeToCalled() { pipeToCalled = true; };
+  // Sets a flag on the WritableStream. Used whenever this WritableStream has been the target
+  // of a pipeTo().
```

Comment:
This case can also be a bit problematic. Consider the following case:

```
const { readable, writable } = socket;
const preamble = getPreambleAsReadableStream();
await preamble.pipeTo(writable, { preventClose: true });
// .. now we wait to read from the stream
```

In this case, the `preamble` might be used to write initial connection handshake / pre-tls stuff to the connection before any reads a performed that would signal that `startTls` should be used. Because of the `preventClose`, the `writable` is not fully consumed and automatically closed when preamble is finished.

Reply:
While this is theoretically possible, the vast majority of protocols have a very short preamble and seemingly no use case for piping it from another stream.

Reply:
granted that it would be rare to see, if ever, but it just seems odd that I can call write as much as I want over and over and never have startTls automatically switch off but calling pipeTo does. As I said above, I'm not blocking but I'd certainly prefer not to go this route. I'd be much happier with an option on connect.